### PR TITLE
Math Magicians: Tests

### PR DESCRIPTION
**Pull request by kkolade**
- Set up React Testing Library.
   - Follow the instructions in the official documentation).
- Write unit tests for the files operate.js and calculate.js using Jest.
- Create unit tests for all React components:
   - Use Jest snapshots to test the components.
   - Use React Testing Library to simulate user interaction.
- Run your full tests suite using Jest. All tests should pass.

### CODE REVIEW

File: src/tests/Quote.test.js
Comment (KanzaTahreem) [inline src/tests/Quote.test.js:1]: - [ ] Good job implementing the user interaction tests for `Quote` 🥇 Please consider writing jest snapshot test for `Quote` component as well according to the project requirement✨ You can use similar to the following for `Quote` like you did for other components 

  > _Use Jest snapshots to test the components_

```
describe('Home component', () => {
  it('renders correctly', () => {
    const { container } = render(<Home />);
    expect(container).toMatchSnapshot();
  });
});
```

**Review by KanzaTahreem — CHANGES_REQUESTED** [review]
# Changes Required 🔁

Hi @team 👋

Well done writing tests for React components! 🥇🥇
**_However, there are few things that still need to be fixed. But you are almost there!_** 💪 🏁

so_close

## Highlights 🎯

There are some excellent points that you have made perfectly. Keep up the great work! 👏
- ✅ Well done with the design
- ✅ Descriptive PR and README file
- ✅ Correct Linters
- ✅ Implemented tests are passing

image

However, there is always room for improvement.

## Required Changes 🛑

> _Please keep in mind that your long hours of work is appreciated. However, the code review process is for improving the code you wrote, and not for criticizing it. 😄_

_Please check the comments under the review._

## Optional suggestions ❕
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

N/A

Cheers and Happy coding!👏👨‍💻💻

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/tests/Quote.test.js
Comment (Cabraham1) [reply src/tests/Quote.test.js:1]: Thank you for your professional response, @KanzaTahreem I appreciate your feedback and I will definitely consider the suggestion made, to add Jest snapshot tests for the Quote component. I am glad that the tests I have implemented so far have passed successfully. Thank you again for your prompt and helpful response.

Comment (KanzaTahreem) [reply src/tests/Quote.test.js:1]: Thank you @Cabraham1, Good to hear this. Let me know if you need any help. Keep going 🏃‍♂️ I wish you success 🥇

File: src/tests/Button.test.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import React from 'react';
+import renderer from 'react-test-renderer';
+import { render, screen } from '@testing-library/react';
+import '@testing-library/jest-dom';
+import Button from '../components/Button';
+
+describe('testing Button', () => {
```
Comment (codecaiine) [inline src/tests/Button.test.js]: - [ ] Good job so far by adding the previous reviewer's requested changes but I think while you were adding the user interaction tests for Quote some errors are coming. Please make sure to fix all the errors in your tests and make them pass without errors like before 👍 

error

**Review by codecaiine — CHANGES_REQUESTED** [review]
Hi @kkolade @Cabraham1 @inakiesk , 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights

- Good implementation tests for Quote ✔️ 

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me (@codecaiine) in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by frank1738 — APPROVED** [review]
Hi Team , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights

✅ Correct use of Gitflow
✅ No linter error
✅ All tests are passing 👍 

## Optional suggestions

N/A

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**